sqlite.go

```diff
@@ -172,23 +181,36 @@ type conn struct {
 	stmts    map[string]*stmt // persisted statements
 	txState  txState
 	readOnly bool
+	closed   atomic.Bool
 }
 
 func (c *conn) Prepare(query string) (driver.Stmt, error) { panic("deprecated, unused") }
 func (c *conn) Begin() (driver.Tx, error)                 { panic("deprecated, unused") }
 func (c *conn) Close() error {
+	// Don't double-close
+	if c.closed.Load() {
+		return nil
+	}
+
 	for q, s := range c.stmts {
 		s.stmt.Finalize()
 		delete(c.stmts, q)
 	}
-	return reserr(c.db, "Conn.Close", "", c.db.Close())
+	err := reserr(c.db, "Conn.Close", "", c.db.Close())
+	c.closed.Store(true)
+	return err
 }
 func (c *conn) PrepareContext(ctx context.Context, query string) (driver.Stmt, error) {
 	persist := ctx.Value(persistQuery{}) != nil
 	return c.prepare(ctx, query, persist)
 }
 
 func (c *conn) prepare(ctx context.Context, query string, persist bool) (s *stmt, err error) {
+	if c.closed.Load() {
+		UsesAfterClose.Add(1)
```

Comment:
should we make it a https://pkg.go.dev/expvar#Map and `.Add("prepare", 1)` ?

Reply:
Yeah, I like that; let me update